aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/DownloadCallable.java

```diff
@@ -197,7 +196,10 @@ private void downloadInParallel(int partCount) throws Exception {
      * Merges all the individual part Files into dstFile
      */
     private void combineFiles() throws Exception {
-        truncateDestinationFileIfNecessary();
+        if (!ServiceUtils.createParentDirectoryIfNecessary(dstfile)) {
```

Comment:
This won't fix the problem. `futureFiles` starts download concurrently, which may happen before this is excuted